app/src/main/java/com/scouts/kitchenplaner/model/usecases/EditRecipe.kt

```diff
@@ -0,0 +1,223 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023-2024 Arne Kuchenbecker, Antonia Heiming, Anton Kadelbach, Sandra Lanz
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+package com.scouts.kitchenplaner.model.usecases
+
+import android.net.Uri
+import com.scouts.kitchenplaner.datalayer.KitchenAppDataStore
+import com.scouts.kitchenplaner.datalayer.repositories.RecipeRepository
+import com.scouts.kitchenplaner.model.entities.DietarySpeciality
+import com.scouts.kitchenplaner.model.entities.Ingredient
+import com.scouts.kitchenplaner.model.entities.IngredientGroup
+import com.scouts.kitchenplaner.model.entities.Recipe
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.onStart
+import java.util.Date
+import javax.inject.Inject
+
+/**
+ * Usecase for editing a recipe
+ *
+ * @param recipeRepository The repository to be used to access the data layer
+ */
+class EditRecipe @Inject constructor(
+    private val recipeRepository: RecipeRepository,
+    private val userDataStore: KitchenAppDataStore
+) {
+    /**
+     * Get a recipe by its ID
+     *
+     * @param id The ID of the requested recipe
+     * @return A flow containing the requested recipe
+     */
+    fun getRecipe(id: Long): Flow<Recipe> {
+        return recipeRepository.getRecipeById(id).onStart {
+            recipeRepository.updateLastShownRecipeForUser(
+                userDataStore.getCurrentUser(),
```

Comment:
(As todo, not for this issue) do we want to update this even if there is no current user?

Reply:
iirc we said that to do anything in the app, you have to be logged in (regardless of whether that would be technically necessary) precisely in order to prevent issues like this where we have to figure out whether there is a current user. If we want to change that assumption, that should be a separate issue in my opinion.